tests/test_utils.py

```diff
@@ -0,0 +1,206 @@
+def test_create_term_list():
+    """Test that only the terms associated with the given vocabulary are returned."""
+    vocab = VocabularyFactory()
+    vocab_terms = TermFactory.create_batch(4, vocab_list=vocab)
+    other_term = TermFactory()
+
+    term_list = views.create_term_list(vocab.id)
+    for term in vocab_terms:
+        assert str(term) in str(term_list)
+    assert str(other_term) not in str(term_list)
+
+
+def test_has_spaces_with_spaces():
+    with pytest.raises(ValidationError):
+        admin.has_spaces(' J o h n ')
+
+
+def test_has_spaces_without_spaces():
+    assert admin.has_spaces('John') == 'John'
+
+
+@pytest.mark.django_db
+class TestVocabularyHandler():
+
+    def test_xml_response(self):
+        vocab = VocabularyFactory()
+        vocab_handler = vocabulary_handler.VocabularyHandler().xml_response(vocab)
+        assert vocab_handler.vocab == vocab
+        assert isinstance(vocab_handler, vocabulary_handler.VocabularyHandler)
+
+    def test_create_xml(self):
+        # XML namespaces.
+        rdf_ns = '{http://www.w3.org/1999/02/22-rdf-syntax-ns#}'
+        dc_ns = '{http://purl.org/dc/elements/1.1/}'
+        rdfs_ns = '{http://www.w3.org/2000/01/rdf-schema#}'
+
+        prop = PropertyFactory(property_name='definition')
+        term = prop.term_key
+        vocab = term.vocab_list
+
+        vocab_handler = vocabulary_handler.VocabularyHandler().xml_response(vocab)
+
+        # Check that all the expected elements are in the vocab_list attribute and that they
+        # have the expected values and attributes.
+        root = objectify.fromstring(vocab_handler.vocab_file)
+        assert root.tag == '{}RDF'.format(rdf_ns)
+        rdf_description = root['{}Description'.format(rdf_ns)]
+        assert rdf_description.get('{}about'.format(rdf_ns)) == (
+            'http://purl.org/NET/UNTL/vocabularies/formats/')
+        assert rdf_description['{}title'.format(dc_ns)] == vocab.label
+        assert rdf_description['{}publisher'.format(dc_ns)] == (
+            'University of North Texas Libraries')
+        assert rdf_description['{}description'.format(dc_ns)] == vocab.definition
+        assert rdf_description['{}language'.format(dc_ns)] == 'English'
+        assert str(rdf_description['{}date'.format(dc_ns)]) == vocab.created.strftime('%Y')
+        rdf_property = root['{}Property'.format(rdf_ns)]
+        assert rdf_property.get('{}about'.format(rdf_ns)) == (
+            'http://purl.org/NET/UNTL/vocabularies/formats/#{}'.format(term.name))
+        assert rdf_property['{}label'.format(rdfs_ns)] == term.label
+        assert rdf_property['{}description'.format(dc_ns)] == prop.label
+        assert rdf_property['{}isDefinedBy'.format(rdfs_ns)].get('{}resource'.format(rdf_ns)) == (
+            'http://purl.org/NET/UNTL/vocabularies/formats/')
+        assert vocab_handler.vocab_mimetype == 'text/xml'
+        assert '<?xml version="1.0" encoding="UTF-8"?>' in vocab_handler.vocab_file
```

Comment:
Yeah this test and a couple others in here are kinda aweful. I'll see if I can come up with something. The only other thing I could think of at the time was creating an string with the full expected xml in it, but I dunno, that seemed to be just as ugly. Not sure.